### PR TITLE
python310Packages: Improve unsupported throws

### PR DESCRIPTION
**Pull request by mweinelt**
by consistently printing the system and package name.

```diff
❯ nix-build -A tts --system aarch64-linux
-error: Unsupported system
+error: Unsupported system aarch64-linux for package torch-bin
```

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by dotlambda** [comment]
Why don't we use `meta.platforms` for this?

**Comment by mweinelt** [comment]
Because only hydra cares for that attribute and when you try to build a package where your system is not part of `meta.platforms` you'll get an evaluation error for the `src`. Not sure if that is better?

```
❯ nix-build --system aarch64-linux -A python3.pkgs.torch-bin
error: attribute 'aarch64-linux-310' missing

       at /home/hexa/git/nixos/master/pkgs/development/python-modules/torch/bin.nix:34:18:

           33|
           34|   src = fetchurl srcs."${stdenv.system}-${pyVerNoDot}";
             |                  ^
           35|
⚠︎ Exited with 1 errors reported by nix at 16:47:00 after 0s
```

**Review by Mindavi — APPROVED** [review]

**Comment by dotlambda** [comment]
> Because only hydra cares for that attribute and when you try to build a package where your system is not part of `meta.platforms` you'll get an evaluation error for the `src`. Not sure if that is better?

That's not true. If `meta.platforms = [ "aarch64-darwin" "x86_64-darwin" "x86_64-linux" ]` then the error message is
```
$ nix-build -A tts --system aarch64-linux
error: Package ‘python3.10-torch-1.13.1’ in /home/rschuetz/devel/nixpkgs/pkgs/development/python-modules/torch/bin.nix:71 is not supported on ‘aarch64-linux’, refusing to evaluate.

       a) To temporarily allow packages that are unsupported for this system, you can use an environment variable
          for a single invocation of the nix tools.

            $ export NIXPKGS_ALLOW_UNSUPPORTED_SYSTEM=1

        Note: For `nix shell`, `nix build`, `nix develop` or any other Nix 2.4+
        (Flake) command, `--impure` must be passed in order to read this
        environment variable.

       b) For `nixos-rebuild` you can set
         { nixpkgs.config.allowUnsupportedSystem = true; }
       in configuration.nix to override this.

       c) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
         { allowUnsupportedSystem = true; }
       to ~/.config/nixpkgs/config.nix.
```

**Comment by mweinelt** [comment]
Indeed, not sure what I tested then.

File: pkgs/development/python-modules/debugpy/default.nix [hunk 1]
```diff
@@ -66,7 +66,7 @@ buildPythonPackage rec {
       "x86_64-darwin"  = "-std=c++11 -lc -D_REENTRANT -dynamiclib -o attach_x86_64.dylib";
       "i686-darwin"    = "-std=c++11 -lc -D_REENTRANT -dynamiclib -o attach_x86.dylib";
       "aarch64-darwin" = "-std=c++11 -lc -D_REENTRANT -dynamiclib -o attach_arm64.dylib";
-    }.${stdenv.hostPlatform.system} or (throw "Unsupported system: ${stdenv.hostPlatform.system}")}
+    }.${stdenv.hostPlatform.system} or (throw "Unsupported system: ${stdenv.hostPlatform.system} for package ${pname}")}
```
Comment (dotlambda) [inline pkgs/development/python-modules/debugpy/default.nix:69]: An "attribute missing" error should be fine if someone sets `NIXPKGS_ALLOW_UNSUPPORTED_SYSTEM=1`.
```suggestion
    }
```

**Comment by dotlambda** [comment]
Btw, this is the beauty of lazy languages :)

File: pkgs/development/python-modules/debugpy/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/debugpy/default.nix:69]: python's disable in python-packages is rather dirty and if we remove the or here we error with a syntax error when the platform is not supported.

Comment (dotlambda) [reply pkgs/development/python-modules/debugpy/default.nix:69]: It not about `disabled`, this is about `meta.platforms`.

**Pull request closed without merging**